modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+        current_volume_set = volume_ids[set_index:set_index + set_count]
+        waiter = client.get_waiter('volume_available')
+        waiter.wait(VolumeIds=current_volume_set)
+        attached = modify_volume_set(
+            client, print, 'attach_volume', instance_id, current_volume_set)
+        if not attached:
+            print(' Volume attachment failed')
+            print(' Exiting...')
+            return False
+        running = input_helper(input)
+        detached = modify_volume_set(
+            client, print, 'detach_volume', instance_id, current_volume_set)
+        if not detached:
+            print(' Volume detachment failed')
+            print(' Exiting...')
+            return False
+        waiter.wait(VolumeIds=current_volume_set)
+        set_index += set_count
+        if not running:
+            break
+    cleanup(client)
+    return True
+
+
+def modify_volume_set(client, print, func, instance_id, volume_id_set):
+    """Helper function to load volumes on an instance to not overload the
+    instance.
+
+    Args:
+        client (boto3.client): client to interact with AWS
+        print (func): Overwritten built-in print function
+        func (str): Function sname to modify_volume_set
+        instance_id (str): instance_id to attach volumes to
+        volume_ids (list): list of volume_ids to (de)attach to the instance.
+    Returns:
+        bool: True if the volumes were successfully attached.
+    """
+    for volume_id in volume_id_set:
+        try:
+            kwargs = {
+                'InstanceId':instance_id,
+                'VolumeId':volume_id
+            }
+            if func == 'attach_volume':
+                kwargs['Device'] = get_valid_device(client, instance_id)
+            caller = getattr(client, func)
+            caller(**kwargs)
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if  code == 'UnauthorizedOperation':
+                print('  FAILURE MISSING AWS PERMISSIONS')
+            else:
+                print(error)
+            return False
+    return True
+
+def get_valid_device(client, instance):
+    """Returns the next device mapping available
+
+    Args:
+        client (boto3.client): Client that gets the current block device mappings
+        instance (str): InstanceId to get curretn block device mappings
+    Returns:
+        str: Returns next mapping in form of /dev/xvd[base], otherwise /dev/xvdzz
+
+    """
+    response = client.describe_instances(InstanceIds=[instance])
+    mappings = response['Reservations'][0]['Instances'][0]['BlockDeviceMappings']
+    current_mappings = [device['DeviceName'] for device in mappings]
+    base_mappings = [char for char in 'bcdefghijklmnoqrstuvwxyz']
+    for base in base_mappings:
+        if '/dev/xvd' + base not in current_mappings:
+            return '/dev/xvd' + base
+    return '/dev/xvdzz'
+
+
+def get_snapshots(pacu, session, region):
```

Comment:
Is this supposed to be region or regions? On line 122 it says it is a list of regions